### ISSUE TITLE
Embedded MPV in java application opens in an external window

### ISSUE THREAD

**Issue by MystEr3-84** [mpv-player/mpv#15543]
### mpv Information

```bash
mpv 0.39.0 Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
libplacebo version: v7.349.0
FFmpeg version: 7.1
FFmpeg library versions:
   libavcodec      61.19.100
   libavdevice     61.3.100
   libavfilter     10.4.100
   libavformat     61.7.100
   libavutil       59.39.100
   libswresample   5.3.100
   libswscale      8.3.100
```

### Other Information

```text
- macOS version: macOS 14.1 (23B2073)
- Source of mpv: homebrew
- Latest known working version: 0.34.1_2 embedded in Java 8 application
- Issue started after the following happened: after migration of the application to Java 17 and updated MPV version
```

### Reproduction Steps

We have a Java application (Eclipse RCP/SWT) that embeds MPV.
Everything worked fine with MPV 0.34.1_2 embedded in Java 8 application.
Now with MPV 0.39 embedded in Java 17 application the MPV opens in an external window.
We found this discussion https://github.com/mpv-player/mpv/discussions/13974 telling about MPV opening in an external window too, but the OS is not specified. We tried the solution `mpv_set_option_string(mpv, "vo", "libmpv")`, on MacOs it doesn't open an external window but there is only the sound and no video.
Does someone know if it is possible to embed MPV in an application on MacOs without MPV opening in an external window?

### Expected Behavior

MPV embedded in the application

### Actual Behavior

MPV opening in an external window

### Log File

[log.log](https://github.com/user-attachments/files/18195163/log.log)

### Sample Files

A little example opening an SWT window (needs Eclipse IDE to be executed), loading MPV with JNA and opening a video.
[MPVEmbedSWT2.txt](https://github.com/user-attachments/files/18186188/MPVEmbedSWT2.txt)

Here you can see the java window on top left corner, and the MPV running to the bottom right.
<img width="1021" alt="2024-12-17 16 25 55" src="https://github.com/user-attachments/assets/045e2c95-8bdb-4547-ab0e-db5e71f33eea" />

### I carefully read all instruction and confirm that I did the following:

- [X] I tested with the latest mpv version to validate that the issue is not already fixed.
- [X] I provided all required information including system and mpv version.
- [X] I produced the log file with the exact same set of files, parameters, and conditions used in "Reproduction Steps", with the addition of `--log-file=output.txt`.
- [X] I produced the log file while the behaviors described in "Actual Behavior" were actively observed.
- [X] I attached the full, untruncated log file.
- [X] I attached the backtrace in the case of a crash.

**Comment by Akemi** [comment]
wid embedding is atm not supported anymore. it was removed with the old deprecated cocoa backend https://github.com/mpv-player/mpv/pull/12861.

see also our macOS todos (https://github.com/mpv-player/mpv/issues/13608), "add wid embedding to macvk".

TLDR:
- you used an old deprecated backend that wasn't maintained anymore since years
- if you still want to use it, you need to use an older (lib)mpv version
- if you want to use `vo=libmpv` you need to implement your own render loop via the libmpv render api (example: https://github.com/mpv-player/mpv-examples/blob/master/libmpv/cocoa-rendergl/cocoa-rendergl.m, api: https://github.com/mpv-player/mpv/blob/master/libmpv/render.h)
- our docs need an update in that regard

closing since it works as intended for now.
